ansible_base/authentication/utils/user.py

```diff
@@ -0,0 +1,33 @@
+from django.contrib.auth.models import AbstractUser
+
+from ansible_base.authentication.authenticator_plugins.utils import get_authenticator_plugin
+from ansible_base.authentication.models import AuthenticatorUser
+from ansible_base.lib.utils.models import is_system_user
+
+
+def can_user_change_password(user: AbstractUser) -> bool:
+    """
+    See if the given user is allowed to change their password.
+    True if they are authenticated from the `local` authenticator
+    False otherwise.
+    SYSTEM_USER can never change their password
```

Comment:
I think it might be worth starting to standardize docstring comments for functions and methods. We don't have to be religious about it, but if we start trying to follow the Sphinx style, then we can generate DAB-internal Python API docs and they'll look really nice. So perhaps:

```suggestion
    Determine if the given user is allowed to change their password.
    
    :param user: The user to check
    :return: `True` if the user is authenticated from the ``local`` authenticator,
        `False` otherwise. Always returns `False` if ``user`` is the system user.
```

Reply:
I 100% agree with this. Maybe we should make an issue/story to do this for all docstrings.